pkg/catalog/inbound_traffic_policies.go

```diff
@@ -304,3 +309,45 @@ func (mc *MeshCatalog) getUpstreamServicesIncludeApex(upstreamServices []service
 
 	return allServices
 }
+
+// getRateLimitServiceClusters returns a list MeshClusterConfig objects corresponding to the global
+// rate limit service instance. It ensures only a single cluster config if the same rate limit service
+// is used for both TCP and HTTP rate limiting.
+func getRateLimitServiceClusters(upstreamTrafficSetting *policyv1alpha1.UpstreamTrafficSetting, clusterSet mapset.Set) []*trafficpolicy.MeshClusterConfig {
+	if upstreamTrafficSetting == nil || upstreamTrafficSetting.Spec.RateLimit == nil || upstreamTrafficSetting.Spec.RateLimit.Global == nil {
+		return nil
```

Comment:
just double checking, it's ok to have a nil cluster?

Tracing through the code looks like there's a few places we can return a nil cluster in `cds.NewResponse`

Reply:
This returns a nil slice, not a nil cluster. So when we do `append(slice, nil_slice...)`, the result is we don't append anything to the original slice.